### PR TITLE
Mrc 1793 use Model Options validation from hintr

### PR DESCRIPTION
**Pull request by carbowizzy**
PR for validation of model options directly from hintr endpoint. It should validate the given options before proceeding to the next page and if validation returns error it should display information regarding the error.

### CODE REVIEW

**Review by EmmaLRussell — CHANGES_REQUESTED** [review]
Looks great, works well. Couple of comments on mutation naming. 
Validation seems to be pretty quick but perhaps in another ticket we should also show a watiing spinner and disable the button while waiting for response, as Calibration does (or will do, once this branch gets merged: https://github.com/mrc-ide/hint/pull/371)

**Review by EmmaLRussell — APPROVED** [review]

File: src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ModelRunTests.kt
Comment (hillalex) [inline src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ModelRunTests.kt:68]: This test doesn't actually test anything - there could be any number of reasons for returning this status, including this endpoint being wrongly wired up!

I'd like to see a test where the response is a 200 and we get back a response that conforms to the `ModelOptionsValidate` schema.

**Review by hillalex — CHANGES_REQUESTED** [review]
We also need a front to bad end integation test - this should live in `modelOptions.itest.ts`.

File: src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ModelRunTests.kt
Comment (hillalex) [inline src/app/src/test/kotlin/org/imperial/mrc/hint/integration/ModelRunTests.kt:68]: Great, thanks. Can we also validate the schema? Something like 
`assertSucccess(responseEntity, "ModelOptionsValidate")`

**Review by hillalex — COMMENTED** [review]
Thanks for this. We still also need a front to back end intergation test - this will live in `modelOptions.itest.ts.`

File: src/app/static/src/tests/integration/modelOptions.itest.ts
Comment (hillalex) [inline src/app/static/src/tests/integration/modelOptions.itest.ts:60]: I'm sorry this is still a bit vague - it could have error-ed because e.g. the route wasn't found. I think checking the error detail will be able to tell us whether the correct endpoint was hit.

File: src/app/static/src/app/store/modelOptions/actions.ts
Comment (hillalex) [inline src/app/static/src/app/store/modelOptions/actions.ts:34]: rogue console log

**Review by hillalex — CHANGES_REQUESTED** [review]

**Review by hillalex — APPROVED** [review]

**Pull request merged**
